packages/neo-one-monitor/src/Reporter.js

```diff
@@ -0,0 +1,165 @@
+  _endpoint: string;
+  _subscription: Subscription;
+  _requestCounter: CounterMetric;
+
+  constructor({
+    logger,
+    metricsFactory,
+    timer,
+    endpoint,
+  }: {|
+    logger: BrowserLogger,
+    metricsFactory: BrowserMetricsFactory,
+    timer: number,
+    endpoint: string,
+  |}) {
+    this._endpoint = endpoint;
+    this._requestCounter = metricsFactory.createCounter({
+      name: 'browser_request_counter',
+      help: 'Counts fetch requests from Browser Reporter to endpoint',
+      labelNames: [KNOWN_LABELS.ERROR],
+    });
+    const emptyBackReport = {
+      logs: [],
+      metrics: {
+        counters: {},
+        histograms: {},
+      },
+    };
+
+    this._subscription = interval(timer)
+      .pipe(
+        mergeScan(
+          async backReport =>
+            this._collectReport({
+              backReport,
+              logger,
+              metricsFactory,
+              emptyBackReport,
+            }),
+          emptyBackReport,
+          1,
+        ),
+      )
+      .subscribe();
+  }
+
+  async _report(report: Report): Promise<Response> {
+    return fetch(this._endpoint, {
+      method: 'POST',
+      body: JSON.stringify(report),
+    });
+  }
+
+  async _collectReport({
+    backReport,
+    logger,
+    metricsFactory,
+    emptyBackReport,
+  }: {|
+    backReport?: Report,
+    logger: BrowserLogger,
+    metricsFactory: BrowserMetricsFactory,
+    emptyBackReport: Report,
+  |}): Promise<Report> {
+    const logs = logger.collect();
+    const metrics = metricsFactory.collect();
+
+    const report = this._addBackReport({ logs, metrics }, backReport);
+
+    let response;
+    try {
+      response = await this._report(report);
+    } catch (error) {
+      response = { ok: false };
+    }
+
+    if (!response.ok) {
+      this._requestCounter.inc({ [KNOWN_LABELS.ERROR]: true });
+      return this._constructBackReport(report);
+    }
+    this._requestCounter.inc({ [KNOWN_LABELS.ERROR]: false });
+    return emptyBackReport;
+  }
+
+  _addBackReport(report: Report, backReport?: Report): Report {
+    if (backReport == null) {
+      return report;
+    }
+    const logs = backReport.logs.concat(report.logs);
+    const currentMetrics = report.metrics;
+
+    const newMetrics = utils.entries(backReport.metrics).reduce(
+      (accMetricType, [metricType, metrics]) => ({
+        ...accMetricType,
+        [metricType]: utils.entries(metrics).reduce(
+          (accMetric, [name, metric]) => ({
+            ...accMetric,
+            [name]: {
+              metric: metric.metric,
+              values:
+                currentMetrics[metricType][name] == null
+                  ? metric.values
+                  : metric.values.concat(
+                      currentMetrics[metricType][name].values,
+                    ),
+            },
+          }),
+          { ...currentMetrics[metricType] },
+        ),
+      }),
+      { counters: {}, histograms: {} },
```

Comment:
I think you want to set this to `currentMetrics`. Otherwise, if there are no metrics in `backReport.metrics` then `newMetrics` will always be this empty object rather than whatever is in `currentMetrics`.

So, change this line to just be `currentMetrics` (or just `report.metrics`) and change line 124 to be just `accMetricType[metricType] || {}`.

Also, a general note - remember, everything is treated as immutable, you don't need to copy everywhere. To be specific, we're always making copies on write (i.e. that's why it's treated as immutable), so we don't need to also make copies on read, e.g. what you're doing on line 124.